webpack.config.js

```diff
@@ -1,364 +1,288 @@
+	mode: process.env.NODE_ENV === "production" ? "production" : "development",
+	entry: {
+		editor: ["./src/index.web.js", "./src/stylesheets/main.scss"],
+	},
+	optimization: {
+		minimizer: [new CssMinimizerPlugin(), new TerserPlugin()],
+	},
 	output: {
-		path: path.join(__dirname, './build'),
-		filename: 'web/editor.js',
-		library: 'zotero-editor',
-		libraryTarget: 'umd',
-		publicPath: '/',
-		umdNamedDefine: true
+		path: path.resolve(__dirname, "./build/web"),
+		filename: "[name].js",
+		library: {
+			name: "zotero-editor",
+			type: "umd",
+			umdNamedDefine: true,
+		},
 	},
 	module: {
 		rules: [
 			{
 				test: /\.(js|jsx)$/,
 				exclude: /node_modules/,
 				use: {
-					loader: 'babel-loader',
-					options: {
-						presets: [
-							'@babel/preset-react',
-							[
-								'@babel/preset-env',
-								{
-									modules: false
-								}
-							]
-						],
-						'plugins': [
-							'@babel/plugin-transform-runtime',
-							'@babel/plugin-proposal-class-properties'
-						]
-					}
-				}
+					loader: "babel-loader",
+				},
 			},
 			{
 				test: /\.scss$/,
 				use: [
+					MiniCssExtractPlugin.loader,
 					{
-						loader: 'file-loader',
-						options: {
-							name: 'web/editor.css'
-						}
+						loader: "css-loader",
 					},
 					{
-						loader: 'extract-loader'
+						loader: "postcss-loader",
 					},
 					{
-						loader: 'css-loader',
-						options: {
-							sourceMap: true,
-							url: false
-						}
+						loader: "sass-loader",
 					},
-					{
-						loader: 'postcss-loader',
-						options: {
-							sourceMap: true
-						}
-					},
-					{
-						loader: 'sass-loader',
-						options: {
-							sourceMap: true
-						}
-					}
-				]
+				],
+			},
+			{
+				test: /\.svg$/,
+				type: 'asset/resource',
+				generator: {
+					filename: 'assets/[name].[ext]',
+				}
 			}
-		]
-	},
-	resolve: {
-		extensions: ['*', '.js']
+		],
 	},
 	plugins: [
-		new CopyWebpackPlugin([
-				{ from: 'res/', to: 'web/' },
-				{ from: 'html/editor.web.html', to: 'web/editor.html' }
-			], { copyUnmodified: true }
-		),
-		new WriteFilePlugin()
-	]
+		new CleanWebpackPlugin(),
+		new MiniCssExtractPlugin({
+			filename: "[name].css",
+		}),
+		new HtmlWebpackPlugin({
+			template: "./html/editor.web.html",
+			filename: "./[name].html",
+		}),
+	],
 };
 
 const configZotero = {
-	mode: process.env.NODE_ENV === 'production' ? 'production' : 'development',
-	devtool: false,
-	entry: [
-		'./src/index.zotero.js',
-		'./src/stylesheets/main.scss'
-	],
-	output: {
-		path: path.join(__dirname, './build/zotero'),
-		filename: 'editor.js',
-		library: 'zotero-editor',
-		libraryTarget: 'umd',
-		publicPath: '/',
-		umdNamedDefine: true
+	name: "zotero",
+	mode: process.env.NODE_ENV === "production" ? "production" : "development",
+	entry: {
+		editor: ["./src/index.zotero.js", "./src/stylesheets/main.scss"],
 	},
 	optimization: {
-		minimize: false
+		minimizer: [new CssMinimizerPlugin(), new TerserPlugin()],
```

Comment:
We should not minimize code for the Zotero build to have a proper stack trace in Zotero client debug log. Unless we can figure out how Zotero client could use source mapping to report the correct line.

You can see the issue by throwing an error anywhere in note-editor source (i.e. when initialising it).